crates/fluvio-benchmark/src/benchmark_driver.rs

```diff
@@ -1,246 +1,124 @@
-
-            // Clean up the batch
-            debug!("Cleaning up batch");
-            send_control_message(
-                &mut tx_controls,
-                ControlMessage::CleanupBatch {
-                    produce_stats: i != 0,
-                },
-            )
-            .await?;
-            expect_success(&mut rx_success, &config, num_expected_messages).await?;
+        println!("benchmark started");
 
-            // Wait between batches
-            debug!(
-                "Waiting {:?} between samples",
-                config.duration_between_samples
-            );
+        // sleep every second
+        let max_run = 20;
 
-            let elapsed = now.elapsed();
-            sleep(config.duration_between_samples).await;
+        // delay 1 seconds, so produce can start
+        sleep(std::time::Duration::from_secs(1)).await;
 
-            if i != 0 {
-                info!(
-                    "Sample {} / {} complete, took {:?} + {:?}",
-                    i, config.num_samples, elapsed, config.duration_between_samples
-                );
-            }
-        }
-        // Close all worker tasks.
-        send_control_message(&mut tx_controls, ControlMessage::Exit).await?;
-        for jh in workers_jh {
-            timeout(config.worker_timeout, jh).await???;
+        for _run in 0..max_run {
+            // get current stats
+
+            let run_start_num_messages = stat.message_send.load(Ordering::Relaxed);
+            let run_start_bytes = stat.message_bytes.load(Ordering::Relaxed);
+
+            let start_time = std::time::Instant::now();
+            sleep(std::time::Duration::from_secs(2)).await;
+            let elapse = start_time.elapsed().as_millis();
+
+            let run_end_bytes = stat.message_bytes.load(Ordering::Relaxed);
+            let run_end_num_messages = stat.message_send.load(Ordering::Relaxed);
+            let bytes_send = run_end_bytes - run_start_bytes;
+            let message_send = run_end_num_messages - run_start_num_messages;
+
+            println!("total bytes send: {}", bytes_send);
+            println!("total message send: {}", message_send);
+            let bytes_per_sec = (bytes_send as f64 / elapse as f64) * 1000.0;
+            let human_readable_bytes = format!("{:9.1}mb/s", bytes_per_sec / 1000000.0);
+            let message_per_sec = ((message_send as f64 / elapse as f64) * 1000.0).round();
+            println!(
+                "message: per second: {}, bytes per  sec: {}, ",
+                message_per_sec, human_readable_bytes
+            );
         }
 
+        stat.end.store(true, Ordering::Relaxed);
+
         Ok(())
     }
-    pub async fn run_benchmark(config: BenchmarkConfig, all_stats: AllStatsSync) -> Result<()> {
+    pub async fn run_benchmark(config: BenchmarkConfig) -> Result<()> {
         // Create topic for this run
 
         let new_topic = TopicSpec::new_computed(config.num_partitions as u32, 1, None);
```

Comment:
Maybe we create the topic inside the producer, but no block, I can do this in the next PR